### PR TITLE
eagle: 7.7 -> 9.1.2

### PR DESCRIPTION
**Pull request by lrittel**
Updated eagle to version to 9.1.2 while keeping version 7.7 as eagle7.

###### Motivation for this change

Eagle was _a bit_ outdated (see #43695).

###### Things done

- [X] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by bjornfor** [comment]
I'm getting this runtime error:

```
$ nix-build -A eagle
/nix/store/z8pm3csczwy830z16qzrqg3igcs5ql9z-eagle-9.1.2
$ ./result/bin/eagle 
terminate called after throwing an instance of 'std::runtime_error'
  what():  locale::facet::_S_create_c_locale name not valid
Aborted
```

Tried with LANG=C (wild guess) but didn't change anything.

File: pkgs/applications/science/electronics/eagle/eagle.nix [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+      cp "$desktopItem"/share/applications/* "$out"/share/applications/
+      mkdir -p "$out"/share/icons
+      ln -s "$out/eagle-${version}/bin/eagle-logo.png" "$out"/share/icons/eagle.png
+    '';
+  
+    meta = with stdenv.lib; {
+      description = "Schematic editor and PCB layout tool from Autodesk (formerly CadSoft)";
+      homepage = https://www.autodesk.com/products/eagle/overview;
+      license = licenses.unfree;
+      platforms = platforms.linux;
```
Comment (volth) [inline pkgs/applications/science/electronics/eagle/eagle.nix]: ```platforms = [ "x86_64-linux" ]``` ?

**Comment by rittelle** [comment]
I got the same error when I tried to build eagle with nixpkg from Github like:
```
nix-shell -Inixpkgs=$PATH_TO_NIXPKGS_MASTER -p eagle
```
Using this instead fixed it for me:
```
nix-build -E 'with import <nixpkgs> { }; libsForQt5.callPackage ./eagle.nix {}'
```
My guess was that you mix incompatible glibc versions (see #38991), master seems to be at 2.27, stable (on my machines) on 2.26.
This is not an eagle issue and can be worked around by using the packages from stable to build eagle (the `nix-build` command from above).

**Comment by rittelle** [comment]
Okay, I think I applied the suggestion of @volth.
Is there a way to notify the users of this package that the old version is still available as `eagle7`?
Eagle is unable to open example projects, working on my own projects works though.
Maybe I'll find a fix later, but this should be a minor issue (maybe it is a good idea to also create one here?).

**Comment by Mic92** [comment]
Why would you still keep the old version?

**Comment by bjornfor** [comment]
I believe eagle >= 8.x requires an account and a subscription to use it. Older version is something you buy and own yourself.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -20805,7 +20805,9 @@ with pkgs;
 
   adms = callPackage ../applications/science/electronics/adms { };
 
-  eagle = callPackage ../applications/science/electronics/eagle { };
+  eagle7 = callPackage ../applications/science/electronics/eagle/eagle7.nix { };
```
Comment (Mic92) [inline pkgs/top-level/all-packages.nix]: Can you add this as an explanation here?

Comment (Mic92) [reply pkgs/top-level/all-packages.nix]: Have you checked, that the old version is still functional?

**Comment by rittelle** [comment]
The old version is still functional, but I have not tried, to run both versions at the same time.
Eagle versions ≥8.0 require an Autodesk account and the pricing changed from single payment to subscription like @bjornfor said.
I assumed that either people are not upgrading deliberately (like @bjornfor maybe?) or that it is not really used on Nix.
I personally don't mind dropping Eagle 7.
I'm going to add a comment later today.

**Comment by Mic92** [comment]
If it was not upgraded because of lack of maintenance adding more versions is not a good idea.

**Comment by bjornfor** [comment]
I'm not an active eagle user (never was), but I would like to keep the 7.x version around for the single reason that it can be run without subscription. I didn't update/add eagle >= 8.x due to lack of time/interest and dislike of the new subscription model. See my original reasoning in 28f780b3200bf7abdbc5b985cb255a908e46faf4.

**Comment by Mic92** [comment]
Ok. @rittelle please add an explanation why we keep 7.x around.

**Comment by rittelle** [comment]
@Mic92 like this?
I'd appreciate corrections if it contains spelling or grammar errors.
@bjornfor I removed you from the maintainer list of Eagle 9 based on your responses above (the lack of interest), so you are not listed as a maintainer for a package you never used ;) .

**Comment by bjornfor** [comment]
I'm still getting that fatal startup error:

```terminate called after throwing an instance of 'std::runtime_error'
  what():  locale::facet::_S_create_c_locale name not valid
Aborted
```

Wasn't that fixed? I'm on NixOS 18.03.

**Comment by rittelle** [comment]
How do you build and run the application?
Have you seen how I got it running on my system (also 18.03 stable) and my guess why it is not working in a comment above ([here](https://github.com/NixOS/nixpkgs/pull/44951#issuecomment-412471925))?

**Comment by Mic92** [comment]
I do do not have that problem with `nix-review pr 44951`. @bjornfor can you reproduce the problem there?

**Comment by bjornfor** [comment]
@rittelle: I build and run with ```git checkout origin/pr/44951 && nix-build -A eagle && ./result/bin/eagle```.

Using  ```nix-build -E 'with import <nixpkgs> { }; libsForQt5.callPackage ./eagle.nix {}'``` is not reproducible since ```<nixpkgs>``` resolves to different things depending on what's in $NIX_PATH (different glibc, qt etc.). When I'm on NixOS 18.03 it works. When I switch to unstable/master it fails, in the same way as running ```nix-build -A eagle && ./result/bin/eagle``` in the nixpkgs tree.

@Mic92: Yes, I still get the problem when using ```nix-review pr 44951``` and invoking ```eagle``` in its nix-shell. How weird that we get different result.

**Comment by bjornfor** [comment]
I found what's wrong. The $LOCALE_ARCHIVE on NixOS 18.03 is too old. It works when the application is wrapped with ```LOCALE_ARCHIVE=${glibcLocales}/lib/locale/locale-archive``` when ```${glibcLocales}``` is from master.

It appears $LOCALE_ARCHIVE is set globally on NixOS. I don't know if there is a policy against wrapping applications with $LOCALE_ARCHIVE, but in this case it would definitely fix an impurity.

**Comment by Mic92** [comment]
This looks to me, like it something should be rather solved in stdenv, then in every package.

**Pull request merged**